### PR TITLE
Add `ignore_if` parameter on Field class

### PR DESCRIPTION
**Pull request by Snotpus**
I ran into a use case where I had an Excel file that had a prepopulated value (of 'No') for all rows for a given Excel table. This value would be updated to 'Yes' in certain cases, but it left me in a weird position where I wanted to consider these extra rows as 'empty' when a given value is present. This solution allows one to add a value, or list of values, as an ignore_if parameter when defining the fields of a layout.

### CODE REVIEW

**Comment by Mikuana** [comment]
The purpose makes sense, and I think you've got a nice solution to implement it, but I'm wondering whether we *should* do this. Conditionally ignoring data feels like it's making this tool too subservient to weird implementations of data entry. I feel like the validation of a data set should ignore the source of that data as much as possible. If we start conditionally ignoring things, we could end up in a strange situation where we **do** ignore a field based upon a conditional value that comes from Excel, but we **dont** ignore a field with that same value if it comes from a database (because we trust the database). If we specify these exceptions at the field level, then we would need to write two separate layouts for what is supposed to be the same data set. When I think about it from that perspective, it makes me think the field and layout layer of the package is not the right place to do this.

Maybe we should think about some kind of injectable rule set that gets applied to the `_BaseFile` subclass? It would ultimately do the same thing that you're doing in this PR, but it would clearly separate the coding that is done for the file/process from the actual data layout itself.

image
Also, it turns out you can copy-pasta images into GitHub.

File: rumydata/field.py [hunk 1]
```diff
@@ -37,10 +37,13 @@ class Field(_BaseSubject):
     :param strip: (optional) apply pre-processing to values in the field which
         applies the str.strip method, removing leading and trailing whitespaces,
         prior to checking rules.
+    :param ignore_if: (optional) specify values which will force the field to be
+        treated as an Ignore field. Can be a single value or list of values
     """
 
     def __init__(self, nullable=False, rules: list = None, **kwargs):
         self.strip = kwargs.pop('strip', None)
+        self.ignore_if = kwargs.pop('ignore_if', None)
```
Comment (Snotpus) [inline rumydata/field.py:46]: `self.ignore_if` = ''

Comment (Snotpus) [reply rumydata/field.py:46]: Establish this empty attribute here, update _BaseFile to pass the values from the file construction...

```
layout = {
    'col1': None,
    'col2': None,
    'col3': None
}

ExcelFile(layout, exceptions = {
    'col1': ['No']
})
```

**Comment by Snotpus** [comment]
I think these last changes do what we want :)

**Comment by Mikuana** [comment]
Merging into new branch so we can skip automated tests.

**Pull request merged**